radix-engine/src/engine/kernel.rs

```diff
@@ -732,11 +643,31 @@ where
                     ));
                     next_frame_node_refs.insert(package_node_id, package_node_pointer);
                 }
+                RENodeId::Proof(..) => {
```

Comment:
I imagine, in the long run, these references will be dynamically expanded when `read_substate` or `borrow_node`?

Reply:
yup exactly. But not on `borrow_node` only on `borrow_substate`

Reply:
I get your point. I mean if `borrow_node` automatically reads the primary substate.